apps/studio/src/components/ResourceSelector/ResourceSelector.tsx

```diff
@@ -1,157 +1,220 @@
-import { Suspense, useState } from "react"
-import {
-  Box,
-  Flex,
-  HStack,
-  Icon,
-  Skeleton,
-  Spacer,
-  Text,
-} from "@chakra-ui/react"
+import { Suspense, useEffect, useState } from "react"
+import { Box, Flex, HStack, Skeleton, Spacer, Text } from "@chakra-ui/react"
 import { Button, Link } from "@opengovsg/design-system-react"
-import { ResourceType } from "~prisma/generated/generatedEnums"
+import { ResourceType } from "@prisma/client"
 import { BiHomeAlt, BiLeftArrowAlt } from "react-icons/bi"
 
+import type { PendingMoveResource } from "~/features/editing-experience/types"
+import { useQueryParse } from "~/hooks/useQueryParse"
+import { sitePageSchema } from "~/pages/sites/[siteId]"
 import { trpc } from "~/utils/trpc"
 import { ResourceItem } from "./ResourceItem"
 
+interface ResourceSelectorProps {
+  queryFn:
+    | typeof trpc.resource.getChildrenOf.useInfiniteQuery
+    | typeof trpc.resource.getFolderChildrenOf.useInfiniteQuery
```

Comment:
comment: here, we depend very strictly on the return type of these 2 specific functions and nexte time, to render, for example, collection pages only, we'd need to keep adding the type here. what happens if the types don't quite conform (eg: 1 query fn needs x prop but another needs y) - i think we should definitely specify a function type here as `queryFn: (input: string) => Data` rather than depending on the trpc type